pkg/webservice/handlers.go

```diff
@@ -236,3 +236,49 @@ func getNodeJSON(nodeInfo *cache.NodeInfo) *dao.NodeDAOInfo {
 		Schedulable: nodeInfo.IsSchedulable(),
 	}
 }
+
+func GetApplicationHistory(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+
+	if imHistory != nil {
+		var result []*dao.ApplicationHistoryDAOInfo
+		records := imHistory.GetRecords()
+		for _, record := range records {
+			if record != nil {
```

Comment:
Can we have records that are not nil after the first nil record? I think not, but please check.
If we cannot why not use a break to jump out of the loop:
```
if record == nil {
	break
}
```

Reply:
Fixed this, but I think we're relying too much on the actual implementation of the `internalMetricsHistory`, and it makes the code too fragile if it is modified in the future.